### ISSUE TITLE
CM13 + 3.0.2-1 + Aquaris M5 -> wrong TWRP password

### ISSUE THREAD

**Issue by Bondarz** [TeamWin/Team-Win-Recovery-Project#699]
Hello,

i have a "bq Aquaris M5" aka "piccolo"
CM13  cm-13.0-20160531-UNOFFICIAL-piccolo.zip
https://s.basketbuild.com/devs/eloimuns/bq/piccolo/
TWRP 3.0.2-1 

after i encrypted my device TWRP requires a password.
i used a PIN Code, not a pattern. also a displaylock with a PIN.
the PINs are 100% correct.

both PINs wont work for TWRP... any ideas?
https://github.com/TeamWin/Team-Win-Recovery-Project/issues/596
could this patch work also for my device?

thanks!

**Comment by Kra1o5** [comment]
Hi, I never try to encrypt with TWRP, @eloimuns @stucki @cmorlok you try this?

Greetings

**Comment by Bondarz** [comment]
i started to encrypt inside from CM13. Now TWRP needs a password. greetings

**Comment by Kra1o5** [comment]
Can you provide a log of TWRP?

**Comment by Bondarz** [comment]
http://pastebin.com/Eig9DFp3 hope this is correct, found the log in "cache/recovery"

**Comment by Kra1o5** [comment]
Thanks, I will check asap.

**Comment by stucki** [comment]
@Kra1o5 I'm not using encryption so I cannot tell. Can you check if this patch is included in your build? https://gerrit.omnirom.org/#/c/17716
Update: Also check for this patch: https://gerrit.omnirom.org/#/c/17933/

**Comment by Kra1o5** [comment]
@stucki , 

Yeah, is included, maybe we need to add keymaster firmware to recovery.

Maybe this weekend I can check it.

**Comment by Bondarz** [comment]
can i update the new (cm-13.0-20160612-UNOFFICIAL-piccolo.zip) via OTA? normaly i flash all my updates with twrp. no problem, i can wait. hope we can fix it without a factoryreset :)

greetings and thanks!

**Comment by Kra1o5** [comment]
In theory yes, you can.

Greetings

**Comment by Bondarz** [comment]
OK, with the OTA Update i have the same problem. After the download and click on update, the phone tried to reboot into the recovery. still a password request from twrp.

greetings

**Comment by wxyzh** [comment]
the same problem on onelpuse one (bacon),my rom is cm-13.0-ZNH2K-20160610-STABLE-Sultan-bacon.zip.
i encrypted my device whith Pattern. when i choose Require Pattern to start device ,TWRP requires Pattern .
but when i disable Secure start-up.,and boot to TWRP, TWRP requires input a password and no way to input Pattern

**Comment by StephenKing** [comment]
I've been following these decryption issues for a while - since I encrypted my M5 using PIN. It's also not possible for me to decrypt / update using encryption from CM13 from 6th of March.

You find my `recovery.log` [here](https://gist.github.com/StephenKing/975a09c5dd23152defcc529beedd6ffb).

@stucki my dear friend, would it be easy for you to build a TWRP including the patch from #596 for me? I could give that a try.

**Comment by Kra1o5** [comment]
Hi,

I will try to do one test build today.

Greetings!

**Comment by Kra1o5** [comment]
Hi @StephenKing @Bondarz @stucki ,

Here a test build: https://s.basketbuild.com/filedl/devs?dev=Kra1o5&dl=Kra1o5/bq/piccolo/Test/TWRP-piccolo_decryptt1.img

Please, open new one here: https://github.com/TeamWin/android_device_bq_piccolo/issues

Greetings

**Comment by StephenKing** [comment]
Thanks for the build. Unfortunately, it still does not work with "Unable to mount storage" error. Created TeamWin/android_device_bq_piccolo#1.

**Comment by stucki** [comment]
Any update here? Does this problem still exist?

**Comment by StephenKing** [comment]
Sure.

**Comment by Bondarz** [comment]
Yes, it`s still not fixed on my phone :P

greetings and thanks guys

**Comment by eloimuns** [comment]
We will check it asap

**Comment by 1tft** [comment]
Hello, problem still exists with piccolo cm-13.0-20160820 and TWRP 3.0.2-1.
@Bondarz Did you ever tried an earlier TWRP version, 2.8.7.0? Did it work?

**Comment by StephenKing** [comment]
Please continue in TeamWin/android_device_bq_piccolo#1.

After the discussion there, I was able to fix it with the custom build and after upgrading my device firmware. See [this comment](https://github.com/TeamWin/android_device_bq_piccolo/issues/1#issuecomment-235716485).

**Comment by stucki** [comment]
Can this be closed?

**Comment by Kra1o5** [comment]
Yes @stucki , Fixed long time ago.

@Dees-Troy  can you close it?

Greetings
